sdk/core/azure-core-http-netty/src/main/java/com/azure/core/http/netty/NettyAsyncHttpClient.java

```diff
@@ -165,16 +176,59 @@
                 // fileContent.getLength() is always not null in FileContent.
-                // Using NettyOutbound.sendFile(Path, long, long) instead of NettyOutbound.sendFile(Path)
-                // the latter involves unnecessary file length lookup which is redundant with FileContent.
-                return reactorNettyOutbound.sendFile(
-                    fileContent.getFile(), fileContent.getPosition(), fileContent.getLength());
-            } else if (binaryDataContent instanceof StringContent) {
-                return reactorNettyOutbound.sendString(Mono.just(binaryDataContent.toString()));
+                if (restRequest.getUrl().getProtocol().equals("https")) {
+                    // NettyOutbound uses such logic internally for ssl connections but with smaller buffer of 1KB.
+                    return reactorNettyOutbound.sendUsing(
+                        () -> FileChannel.open(fileContent.getFile(), StandardOpenOption.READ),
+                        (c, fc) -> {
+                            if (c.channel().pipeline().get(ChunkedWriteHandler.class) == null) {
+                                c.addHandlerLast("reactor.left.chunkedWriter", new ChunkedWriteHandler());
+                            }
+
+                            try {
+                                return new ChunkedNioFile(
+                                    fc, fileContent.getPosition(), fileContent.getLength(), fileContent.getChunkSize());
+                            } catch (IOException e) {
+                                throw Exceptions.propagate(e);
+                            }
+                        },
+                        (fc) -> {
+                            try {
+                                fc.close();
+                            } catch (IOException e) {
+                                LOGGER.log(LogLevel.VERBOSE, () -> "Could not close file", e);
+                            }
+                        });
+                } else {
+                    // Beware of NettyOutbound.sendFile(Path) it involves extra file length lookup.
+                    // This is going to use zero-copy transfer if there's no ssl
+                    return reactorNettyOutbound.sendFile(
+                        fileContent.getFile(), fileContent.getPosition(), fileContent.getLength());
+                }
+            } else if (binaryDataContent instanceof InputStreamContent) {
+                return reactorNettyOutbound.sendUsing(
+                    binaryDataContent::toStream,
+                    (c, stream) -> {
+                        if (c.channel().pipeline().get(ChunkedWriteHandler.class) == null) {
+                            c.addHandlerLast("reactor.left.chunkedWriter", new ChunkedWriteHandler());
+                        }
+
+                        return new ChunkedStream(stream);
+                    },
+                    (stream) -> {
+                        try {
+                            stream.close();
```

Comment:
Does this close the underlying `InputStream` from `InputStreamContent`? Generally, we've called out that our APIs taking `InputStream` do not close them.

Reply:
yeah. it does. I'll change this.